java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -4154,6 +4197,34 @@ void testMatchesRe() {
     });
   }
 
+  @Test
+  void testMatchesReRegexProg() {
+    RegexProgram regexProg1 = new RegexProgram("\\d+", CaptureGroups.NON_CAPTURE);
+    RegexProgram regexProg2 = new RegexProgram("[A-Za-z]+\\s@[A-Za-z]+", CaptureGroups.NON_CAPTURE);
+    RegexProgram regexProg3 = new RegexProgram(".*", CaptureGroups.NON_CAPTURE);
+    RegexProgram regexProg4 = new RegexProgram("", CaptureGroups.NON_CAPTURE);
+    try (ColumnVector testStrings = ColumnVector.fromStrings("", null, "abCD", "ovér the",
+          "lazy @dog", "1234", "00:0:00");
+         ColumnVector res1 = testStrings.matchesReRegexProg(regexProg1);
+         ColumnVector res2 = testStrings.matchesReRegexProg(regexProg2);
+         ColumnVector res3 = testStrings.matchesReRegexProg(regexProg3);
+         ColumnVector expected1 = ColumnVector.fromBoxedBooleans(false, null, false, false, false,
+           true, true);
```

Comment:
Should the last value be `false`? How come `"00:0:00"` matches `\\d+`?

Reply:
Ah, `match`, not `fullmatch`.

Reply:
Resolved through discussion.